src/vmm/src/logger/logging.rs

```diff
@@ -431,343 +170,189 @@ impl Log for Logger {
-            &format!(
-                "[TEST-INSTANCE-ID:{}:WARN:logger.rs:0] msg\n",
-                crnt_thread_name
-            ),
-        );
+        assert_eq!(LevelFilter::from_str("Warning"), Ok(LevelFilter::Warn));
+        assert_eq!(LevelFilter::from_str("WARNING"), Ok(LevelFilter::Warn));
     }
 
     #[test]
-    fn test_thread_name_custom() {
-        let custom_thread = thread::Builder::new()
-            .name("custom-thread".to_string())
-            .spawn(move || {
-                let logger = Logger::mock_new();
-                let mut reader = logger.mock_init();
-                logger
-                    .set_include_level(false)
-                    .set_include_origin(false, false);
-                logger.set_instance_id("".to_string());
-                logger.mock_log(Level::Info, "thread-msg");
-                validate_log(&mut Box::new(&mut reader), "[custom-thread] thread-msg\n");
-            })
+    fn logger() {
+        // Get temp file path.
+        let file = utils::tempfile::TempFile::new().unwrap();
+        let path = file.as_path().to_str().unwrap().to_string();
+        drop(file);
+
+        // Create temp file.
+        let target = std::fs::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .open(&path)
             .unwrap();
-        let r = custom_thread.join();
-        assert!(r.is_ok());
-    }
-
-    // There was a test `test_static_logger` to cover LOGGER however,
-    // it had conflict with `vmm_config::test::test_itest_init_logger`
-    // which also uses tries to initialize logger and fail with an
-    // already initialized error.
-    // Since vmm_config::test_init_logger() cover the logger init test
-    // with more parameters, test_static_logger() was removed.
-    // This comment is kept here so that we don't try to implement
-    // test_static_logger() again in the future.
 
-    #[test]
-    fn test_error_messages() {
+        // Create logger.
+        let logger = Logger(Mutex::new(LoggerConfiguration {
+            target: Some(target),
+            format: LogFormat {
+                show_level: true,
+                show_log_origin: true,
+            },
+        }));
+
+        // Assert results of enabled given specific metadata.
+        assert!(logger.enabled(&Metadata::builder().level(Level::Warn).build()));
+        assert!(logger.enabled(&Metadata::builder().level(Level::Debug).build()));
+
+        // Log
+        let metadata = Metadata::builder().level(Level::Error).build();
+        let record = Record::builder()
+            .args(format_args!("Error!"))
+            .metadata(metadata)
+            .file(Some("dir/app.rs"))
+            .line(Some(200))
+            .module_path(Some("module::server"))
+            .build();
+        logger.log(&record);
+
+        // Assert calling flush panics.
+        std::panic::catch_unwind(|| {
+            logger.flush();
+        })
+        .unwrap_err();
```

Comment:
nit: could be a `#[should_panic]` test. Why do we need to test this at all though? :o

Reply:
This wouldn't work like it is currently, `#[should_panic]` would prevent testing anything after the panic occurs.

Calling `flush` would need to be moved to the end of the test.

> Why do we need to test this at all though?

It adds a small amount of coverage, there's a very small benefit and no notable downside.